crates/proto/src/op/query.rs

```diff
@@ -159,6 +159,29 @@ impl Query {
     pub fn mdns_unicast_response(&self) -> bool {
         self.mdns_unicast_response
     }
+
+    /// Consumes `Query` and returns it's components
```

Comment:
we should probably document the fields here... Though, for query, should we consider just making the fields `pub`? I don't think there would be a lot of danger in that...

Reply:
It's up to you, the only 'danger' is that you're committing the internals to a public API. But if you want to do that then that's fine with me!

Reply:
Yeah, I agree on keeping a consistent API across the types.

Reply:
I've replicated the XParts api across multiple types